lib/parser/openmp-grammar.h

```diff
@@ -520,76 +487,57 @@ TYPE_PARSER(construct<OmpEndDoSimd>(maybe(construct<OmpNowait>("NOWAIT"_tok))))
 TYPE_PARSER(construct<OmpEndDo>(maybe(construct<OmpNowait>("NOWAIT"_tok))))
 
 // OMP END SECTIONS [NOWAIT]
-TYPE_PARSER(skipStuffBeforeStatement >> "!$OMP "_sptok >> "END"_tok >>
-    "SECTIONS"_tok >>
+TYPE_PARSER(startOmpLine >> "END SECTIONS"_tok >>
     construct<OmpEndSections>(
-        maybe("NOWAIT"_tok >> construct<OmpNowait>()) / endOmpLine))
+        maybe("NOWAIT" >> construct<OmpNowait>()) / endOmpLine))
 
 // OMP SECTIONS
-TYPE_PARSER("SECTIONS" >>
-    construct<OpenMPSectionsConstruct>(
-        Parser<OmpClauseList>{} / endOmpLine, block, Parser<OmpEndSections>{}))
+TYPE_PARSER(construct<OpenMPSectionsConstruct>(
+    OmpConstructDirective("SECTIONS"_tok), block, Parser<OmpEndSections>{}))
```

Comment:
`OmpConstructDirective("SECTIONS"_tok)` will save the source location for directive `SECTIONS` into `OmpClauseList`:
```
| | ExecutionPartConstruct -> ExecutableConstruct -> OpenMPConstruct -> OpenMPSectionsConstruct
| | | OmpClauseList = '**sections**'
| | | Block
| | | | ExecutionPartConstruct -> ExecutableConstruct -> OpenMPConstruct -> OmpSection -> Verbatim = ''
| | | | ExecutionPartConstruct -> ExecutableConstruct -> DoConstruct
```